src/DGtal/shapes/doc/moduleHalfEdgeMesh.dox

```diff
@@ -0,0 +1,650 @@
+faces:
+
+- HalfEdgeDataStructure::halfEdgeIndexFromArc : given an arc or two
+  vertices, returns the corresponding half-edge index (logaristhmic
+  cost in the number of half-edges)
+
+- HalfEdgeDataStructure::halfEdgeIndexFromVertexIndex,
+  HalfEdgeDataStructure::halfEdgeIndexFromFaceIndex,
+  HalfEdgeDataStructure::halfEdgeIndexFromEdgeIndex: returns the index
+  of an half-edge that borders the given vertex, face or edge
+  (constant time operation)
+
+- HalfEdgeDataStructure::halfEdge : given an half-edge index, returns
+  the half-edge itself.
+ 
+Then, each half-edge can give you its associated vertex, face, edge, opposite and next half-edge:
+
+- HalfEdgeDataStructure::HalfEdge::toVertex : the end vertex of this
+  half-edge as an index into the vertex array.
+- HalfEdgeDataStructure::HalfEdge::face : index of the associated
+  face into the face array.
+- HalfEdgeDataStructure::HalfEdge::edge : index of the associated edge into the edge array.
+- HalfEdgeDataStructure::HalfEdge::opposite : index of the opposite
+  half-edge into the half-edge array.
+- HalfEdgeDataStructure::HalfEdge::next index of the next half-edge
+  along the face into the half-edge array.
+
+\image html half-edge-data-structure.png "Illustration of half-edge data structure: each edge corresponds to two half-edges with opposite direction. You have access to the next half-edge along the face or to your opposite half-edge. You can also retrieve your associated face or vertex."
+
+Finally an half-edge data structure can give you all the necessary
+neighboring information, and can also list the vertices and arcs that
+lie on the boundary of the data structure:
+
+- HalfEdgeDataStructure::neighboringVertices,
+  HalfEdgeDataStructure::getNeighboringVertices : returns the sequence
+  of vertices that are neighbors to the given vertex. The order of
+  neighbors has meaning and corresponds to the order given at the
+  initialization per triangle.
+
+- HalfEdgeDataStructure::neighboringFaces,
+  HalfEdgeDataStructure::getNeighboringFaces : returns the sequence of
+  faces that are incident to the given vertex.  The order of
+  neighbors has meaning and corresponds to the order given at the
+  initialization per triangle.
+
+- HalfEdgeDataStructure::boundaryVertices,
+  HalfEdgeDataStructure::boundaryHalfEdgeIndices,
+  HalfEdgeDataStructure::boundaryArcs : returns the vertices and/or
+  arcs lying on the boundary of the combinatorial surface (in no
+  particular order).
+
+\subsection HEM_sec1_3 Details about internal representation
+
+It is worthy to note the following elements in this representation:
+
+- half-edges, vertices, edges and faces are numbered consecutively from 0.
+- vertices and faces (here triangles) keep their numbering given at initialization.
+- given a vertex index, an edge index, or a face index, you can get
+  an half-edge incident to it in constant time (stored as a vector).
+- given an arc (i.e. a couple of vertices), you can get the
```

Comment:
ok here is the def of arc. please discuss about it earlier.

Reply:
are arcs oriented edges ?

Reply:
Since an arc is a couple of vertices, yes they are oriented edges.